contracts/feature/TokenBundle.sol

```diff
@@ -4,5 +4,47 @@ pragma solidity ^0.8.0;
 import "./interface/ITokenBundle.sol";
 
 abstract contract TokenBundle is ITokenBundle {
+    uint256 public nextTokenIdToMint;
+    mapping(uint256=>BundleInfo) public bundle;
+
+    function getTokenCount(uint256 tokenId) public view returns (uint256) {
+        return bundle[tokenId].count;
+    }
+
+    function getToken(uint256 tokenId, uint256 index) public view returns (Token memory) {
+        return bundle[tokenId].tokens[index];
+    }
+
+    function getUri(uint256 tokenId) public view returns (string memory) {
+        return bundle[tokenId].uri;
+    }
     
+    function _getNextTokenId() internal returns (uint256 nextTokenId) {
```

Comment:
See the comments for token ID handling w.r.t `nextTokenIdToMint`.

Since we want to hand off token Id handling to the derived contract e.g. Multiwrap, a function like `_getNextTokenId()` should be exposed as a virtual function, so that the derived contract can override it and supply the right values to this function. What do you think about that?

One question is -- conceptually, does `TokenBundle` have anything to do with "token IDs"? Why can't `_getNextTokenId` be `_getIdForNewBundle`? That is, let the user choose whether the identifier of a bundle is a token Id or not.

Think of the potential use cases for `TokenBundle`. I don't think it's limited to contracts like `Multiwrap` that issue NFTs (i.e. token IDs) for a token bundle.

I can see `TokenBundle` being used in a simple escrow contract, or even in a DeFi contract, since the main utility that i see `TokenBundle` providing is a generic + compressed way of describing a list of tokens. This has a wide range of uses.